Servers/UI/OJS.Servers.Ui/Controllers/Api/SubmissionsController.cs

```diff
@@ -55,21 +55,41 @@ public async Task<IActionResult> GetForProfile()
             .MapCollection<SubmissionForProfileResponseModel>()
             .ToOkResult();
 
+    /// <summary>
+    /// Gets a subset of submissions by specific problem and user and given take count.
+    /// </summary>
+    /// <param name="problemId">The id of the problem</param>
+    /// <param name="userId">The id of the user that we want to get the problem submissions for</param>
+    /// <param name="isOfficial">Return from compete or practice participation types</param>
+    /// <param name="take">Number of submissions to return</param>
+    /// <returns>A collection of submissions for a specific problem by user</returns>
+    [HttpGet("{problemId:int}/{userId}")]
+    [ProducesResponseType(typeof(IEnumerable<SubmissionResultsResponseModel>), Status200OK)]
+    public async Task<IActionResult> GetSubmissionResultsByProblemAndUser(
+        int problemId,
+        string userId,
+        [FromQuery] bool isOfficial,
+        [FromQuery] int? take)
+        => await this.submissionsBusiness
+            .GetSubmissionResultsByProblemAndUser(problemId, isOfficial, userId, take)
+            .MapCollection<SubmissionResultsResponseModel>()
+            .ToOkResult();
+
     /// <summary>
     /// Gets a subset of submissions by specific problem and given take count.
     /// </summary>
-    /// <param name="id">The id of the problem</param>
+    /// <param name="problemId">The id of the problem</param>
     /// <param name="isOfficial">Should the submissions be only from compete mode</param>
     /// <param name="take">Number of submissions to return</param>
     /// <returns>A collection of submissions for a specific problem</returns>
     [HttpGet("{id:int}")]
     [ProducesResponseType(typeof(IEnumerable<SubmissionResultsResponseModel>), Status200OK)]
     public async Task<IActionResult> GetSubmissionResultsByProblem(
-        int id,
+        int problemId,
```

Comment:
Is this mapped correctly? The URL is matching "{id:int}"

Reply:
It was a mistake, which is now fixed.